### PR TITLE
Add eval metrics file

### LINKED ISSUE #63: Report performance metrics for GLM & EALSTM
Need new files?

From Sam:
> Are you reporting performance metrics for any of these models? Seems like that would be necessary to know how to use the models/know how much to trust them.

### PR DESCRIPTION
**Pull request by lindsayplatt**
Fixes #63. Code to combine and create new model evaluation metrics file with all three models in one. Output is on Caldera in this repo at `out_data/lake_temp_model_evaluation_metrics.csv`. Below is a quick view of what the output looks like for @AndyMcAliley to be able to write the attribute/entity info.

```r
library(tidyverse)
read_csv("out_data/lake_temp_model_evaluation_metrics.csv") %>% 
    slice(c(1:3, 6:8, 11:13))

  model   driver     season    n_obs n_lakes   bias  RMSE
  <chr>   <chr>      <chr>     <dbl>   <dbl>  <dbl> <dbl>
1 EA-LSTM NLDAS      overall  453234    1503 0.126   2.70
2 EA-LSTM NLDAS      spring   130413    1116 0.302   2.46
3 EA-LSTM NLDAS      summer   234106    1400 0.0737  2.97
4 GLM     NLDAS      overall 2350373    3688 0.676   2.85
5 GLM     NLDAS      winter    96597     737 0.640   1.56
6 GLM     NLDAS      spring   677513    3631 0.904   2.63
7 GLM     GCM-ACCESS overall  398125    1126 0.190  NA   
8 GLM     GCM-CNRM   overall  398125    1126 0.210  NA   
9 GLM     GCM-GFDL   overall  398125    1126 0.200  NA   
```

This also includes the attribute/entity info that Andy wrote.

### CODE REVIEW

File: in_data/evaluation_metrics_EALSTM.csv
Comment (AndyMcAliley) [inline in_data/evaluation_metrics_EALSTM.csv:1]: I double-checked my bias calculations and found that I was reporting the mean of (observations - predictions) while Hayley is reporting the median of (predictions - observations), based on https://github.com/USGS-R/lake-temperature-process-models/blob/bb45904a17a041fa24c90bd6058fae121400dc43/5_evaluate/src/eval_utility_fxns.R#L161. I changed my bias calcuations to match hers.

File: in_text/text_data_release.yml [hunk 1]
```diff
@@ -2019,6 +2020,65 @@ entities:
...
+      data-min: NA
+      data-max: NA
+      data-units: NA
+    -
+      attr-label: season
+      attr-def: The yearly season during which predictions were made, either
+        "spring" (day of year 60 to 172), "summer" (day of year 173 to 242),
+        "fall" (day of year 243 to 334), "winter" (day of year 335 to 59), or
+        "overall" (all seasons).
+      attr-defs: NA
```
Comment (AndyMcAliley) [inline in_text/text_data_release.yml:2050]: Do we need a reference for using these DOYs to define seasons? I remember seeing that these season definitions are common in meteorology but I don't have a reference for that. I copied the numbers from Hayley's code.

**Review by AndyMcAliley — APPROVED** [review]
I was able to `scmake("out_data/lake_temp_model_evaluation_metrics.csv")` after some R package installation struggles (nice for getting min and max). I left a question about references and made a change to the computed metrics to match GLM metrics better. Read over my changes and then hopefully we're good to go!

File: in_text/text_data_release.yml (hunk 1 above)
Comment (lindsayplatt) [reply in_text/text_data_release.yml:2050]: Good question. @hcorson-dosch-usgs do you have an easy reference for these divisions of seasons?

Comment (hcorson-dosch-usgs) [reply in_text/text_data_release.yml:2050]: No - [this](https://github.com/USGS-R/lake-temperature-process-models/pull/56#discussion_r880516658) is the only reference to the seasonal bins that I've found in our discussion. I think other than Jordan's suggestion for the start of summer, I just picked months for each season.

Comment (lindsayplatt) [reply in_text/text_data_release.yml:2050]: OK, then let's leave `attr-defs` as NA indicating that we made this decisions in this data release. People have the DOY ranges used to define the seasons here, so I don't think we need to justify anything further.

**Pull request merged**